modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -20,17 +20,20 @@ import org.tessellation.sdk.domain.healthcheck.consensus.types.types.RoundId
 import org.tessellation.sdk.domain.healthcheck.services.HealthCheck
 
 abstract class HealthCheckConsensus[
-  F[_]: Clock: GenUUID: Spawn: Ref.Make,
+  F[_]: Async: GenUUID: Ref.Make,
+  K <: HealthCheckKey,
   A <: HealthCheckStatus,
-  B <: ConsensusHealthStatus[A]
+  B <: ConsensusHealthStatus[K, A]
 ](
   clusterStorage: ClusterStorage[F],
   selfId: PeerId,
-  driver: HealthCheckConsensusDriver[A, B],
+  driver: HealthCheckConsensusDriver[K, A, B],
   gossip: Gossip[F]
 ) extends HealthCheck[F] {
+  def allRounds: Ref[F, ConsensusRounds[F, K, A, B]]
+
+  def ownStatus(key: K): F[Fiber[F, Throwable, A]]
```

Comment:
Not sure if instead of `Fiber` we shouldn't use `Deferred`? 🤔

Reply:
`Deferred` will be computed once, but not in the background. `Fiber` is more or less like Deferred that can be computed in the background.